kolibri/utils/conf.py

```diff
@@ -160,3 +160,50 @@ def enable_default_plugins():
 
 # read the config file options in here so they can be accessed from a standard location
 OPTIONS = read_options_file(KOLIBRI_HOME)
+
+
+def save_nginx_conf_include(static_root, nginx_conf=None):
+    """
+    Automatically writes the dynamic Nginx configuration include from Kolibri
+    configuration.
+
+    This function is called from within the DJANGO_SETTINGS_MODULE after it
+    has defined STATIC_ROOT.
+
+    The nginx configuration snippet is loaded with in the main Nginx
+    configuration which contains::
+
+        include /path/to/.kolibri/nginx.conf;
+
+    Note that because we cannot load django.conf.settings from this module
+    (because Django settings depend on Kolibri conf), we have a sort of
+    circular issue with STATIC_ROOT. We cannot fetch settings.STATIC_ROOT
+    unless we start writing the Nginx configuration elsewhere during the load
+    order.
+
+    """
+
+    from kolibri.core.content.utils.paths import get_content_dir_path
+
+    if nginx_conf is None:
+        nginx_conf = os.path.join(KOLIBRI_HOME, nginx_conf)
+
+    with open(nginx_conf, 'w') as nginx_conf_file:
+        configuration = (
+            "# This file is maintained AUTOMATICALLY and will be overwritten\n"
+            "#\n"
+            "# Do not edit this file. If you are using the kolibri-server"
+            "package,\n"
+            "# please write custom configurations in /etc/kolibri/nginx.d/\n"
+            "\n"
+            "location /static {\n"
+            "    alias  {{ static_dir }};\n"
+            "}\n"
+            "location /content {\n"
+            "    alias  {{ content_dir }};\n"
+            "}"
+        ).format(
+            static_dir=static_root,
+            content_dir=get_content_dir_path(),
+        )
+        nginx_conf_file.write(configuration)
```

Comment:
why not writing the PORT here too?

Reply:
Also, to avoid a chicken-egg  problem, I think we should write in this file the path to KOLIBRI_HOME and place it in /etc/kolibri, not in KOLIBRI_HOME.
Currently, the kolibri package is assuming the KOLIBRI_HOME is always in ~/.kolibri. I am not sure this is always true.

Reply:
> Also, to avoid a chicken-egg problem, I think we should write in this file the path to KOLIBRI_HOME and place it in /etc/kolibri, not in KOLIBRI_HOME.

This file is written by the `kolibri` process owned by the user with `KOLIBRI_HOME`. This user does not have permissions to write files in `/etc/`.

Reply:
> why not writing the PORT here too?

I haven't considered this yet. The `listen` directive is quite sensitive because it doesn't allow to be specified several times AFAIK. If there is a way around it, could be to specify a parameter or variable for the rest of the configuration to pick up.

Reply:
> This file is written by the kolibri process owned by the user with KOLIBRI_HOME. This user does not have permissions to write files in /etc/.

Yes, but there are always solutions to that /var/run or some other place with 777 permissions. In any case if we are sure the path is going to be always  ~/.kolibri, this is not needed. But if that's not the case we need a solution.

> I haven't considered this yet. The listen directive is quite sensitive because it doesn't allow to be specified several times AFAIK.

I don't get it, why is it going to be specified several times?  It should be only one, with the port Kolibri is setup to use.

Reply:
> Yes, but there are always solutions to that /var/run or some other place with 777 permissions. In any case if we are sure the path is going to be always ~/.kolibri, this is not needed. But if that's not the case we need a solution.

The environment that launches `kolibri` knows and defines `KOLIBRI_HOME`, for instance by running `source /etc/default/kolibri`. This is already solved in the `kolibri-server` package: https://github.com/learningequality/kolibri-server/pull/4

Reply:
> I don't get it, why is it going to be specified several times? It should be only one, with the port Kolibri is setup to use.

When I say that I haven't investigated something, I think it's best not to discuss details before actually investigating it. This was my way of saying why I didn't include it. You can probably come up with reasons why a configuration directive gets specified several times in a configuration stack.